.github/workflows/nginx.yaml

```diff
@@ -0,0 +1,77 @@
+name: NGINX
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: 'always'
+  RUST_BACKTRACE: '1'
+  NGX_CONFIGURE: |
+    auto/configure
+      --with-compat
+      --with-debug
+      --with-http_realip_module
+      --with-http_ssl_module
+      --with-http_v2_module
+      --with-http_v3_module
+      --with-stream
+      --with-stream_realip_module
+      --with-stream_ssl_module
+      --with-threads
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        nginx-ref:
+          # master
+          - stable-1.26
+
+    env:
```

Comment:
Those probably don't need to be defined for steps other than the actual prove run.